src/Nethermind/Nethermind.Network/PeerManager.cs

```diff
@@ -424,20 +424,37 @@ private void DeactivatePeerIfDisconnected(Peer peer, string reason)
             }
         }
 
-        private void SelectAndRankCandidates()
+        private static ActivePeerSelectionCounter[] _enumValues = InitEnumValues();
+
+        private static ActivePeerSelectionCounter[] InitEnumValues()
         {
-            _currentSelection.PreCandidates.Clear();
-            _currentSelection.Candidates.Clear();
-            _currentSelection.Incompatible.Clear();
-            foreach (ActivePeerSelectionCounter value in Enum.GetValues(typeof(ActivePeerSelectionCounter)))
+            Array values = Enum.GetValues(typeof(ActivePeerSelectionCounter));
+            ActivePeerSelectionCounter[] result = new ActivePeerSelectionCounter[values.Length];
+
+            int index = 0;
+            foreach (ActivePeerSelectionCounter value in values)
             {
-                _currentSelection.Counters[value] = 0;
+                result[index++] = value;
             }
 
+            return result;
```

Comment:
Why not Enum.GetValues(typeof(ActivePeerSelectionCounter)).Cast<ActivePeerSelectionCounter>().ToArray() ? You are allocating in the end.